src/sensor_module.c

```diff
@@ -0,0 +1,232 @@
+/******************************************************************************
+ * Includes
+ ******************************************************************************/
+#include "compass_sim.h"
+#include "nmea.h"
+#include "voltage_sim.h"
+#include "sensor_encoder.h"
+#include "sensor_module.h"
+#include <time.h>
+
+/******************************************************************************
+ * Defines
+ ******************************************************************************/
+extern NMEA_Data_u NMEA_Data;
+extern uint32_t encodeSensorPacket(DataSensorPacket_t* data, uint8_t* buf, uint32_t len);
+extern NMEA_Message_e NMEA_Decode(char c);
```

Comment:
Externs should not live in the module - please place in the appropriate headers.